tsdb/engine/tsm1/engine.go

```diff
@@ -2166,6 +2167,17 @@ func (e *Engine) compact(wg *sync.WaitGroup) {
 						level3Groups = level3Groups[1:]
 					}
 				case 4:
+					// This is a heuristic. 100_000 points per block is suitable for when we have a
+					// single generation with multiple files at max block size under 2 GB.
+					if genLen == 1 {
+						// Log TSM files that will have an increased points per block count.
+						for _, f := range level4Groups[0] {
+							e.logger.Info("TSM optimized compaction on single generation running, increasing total points per block to 100_000.", zap.String("path", f))
+						}
+						e.Compactor.Size = tsdb.DefaultMaxPointsPerBlock * 100
```

Comment:
I feel like `tsdb.DefaultMaxPointsPerBlock * 100` should be its own named constant (`tsdb.AgressiveMaxPointsPerBlock`), or the `100` should be a constant (e.g. `tsdb.AgressiveMaxPointsPerBlockMultipler`). The result of an operation between a named constant and a magic number is a magic number, unless the result is assigned to a new named constant.